src/coreclr/vm/peassembly.cpp

```diff
@@ -334,8 +358,10 @@ void PEAssembly::ConvertMDInternalToReadWrite()
         {
             EX_THROW(EEMessageException, (hr));
         }
-        if (pNew == pOld)
+
+        if ((m_pMDImport == m_pConvertedMDImport && pNew == pOld) || !swapForRWMDImport)
         {
+            _ASSERTE(m_pConvertedMDImport != NULL);
```

Comment:
What is this assert telling me?

Reply:
We know from above that `m_pConvertedMDImport` is not `NULL` or else it would have entered the branch above and returned, no?

Reply:
After thinking about it for a bit - it wouldn't really give any valuable information. It verified that in the case an importer exists, we at least have cached the RW in case its raced with the if-condition above. But given that this method caches and it's needed to create the importer, I am not sure it provides much value.